### ISSUE TITLE
Running the AUT in debug mode with `xcodebuild` doesn't work 

### ISSUE THREAD

**Issue by bartekpacia** [leancodepl/patrol#829]
Reproduction:

**Step 0**

```console
cd ./packages/patrol/example/ios
```

**Step 1**

```console
flutter build ios --config-only --target integration_test/open_app_test.dart --debug
```

**Step 2**

```console
xcodebuild test \
  -workspace Runner.xcworkspace \
  -scheme Runner \
  -xcconfig Flutter/Release.xcconfig \
  -configuration Release \
  -sdk iphoneos -destination "platform=iOS,name=Barteks-iPad" \
  OTHER_SWIFT_FLAGS='$(inherited) -D PATROL_ENABLED'
```

**Step 3**

```console
idevicesyslog | tee ~/Desktop/syslog.txt
```

While browsing through that `syslog.txt`, I found:

```
...more logs...
Jan 27 13:31:06 Runner(Flutter)[609] <Notice>: [VERBOSE-2:ptrace_check.cc(75)] Could not call ptrace(PT_TRACE_ME): Operation not permitted
...a few lines...
Jan 27 13:31:06 Runner(Flutter)[609] <Notice>: Cannot create a FlutterEngine instance in debug mode without Flutter tooling or Xcode.

To launch in debug mode in iOS 14+, run flutter run from Flutter tools, run from an IDE with a Flutter IDE plugin or run the iOS project from Xcode.
Alternatively profile and release mode apps can be launched from the home screen.
...more logs...
```

We're hitting this condition in FlutterEngine code:
https://github.com/flutter/engine/blob/main/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm#L416-L430

Flutter app runs with JIT in debug mode. Since iOS 14, apps can only run code JIT when started through developer tools (`flutter run` or from Xcode UI directly). **Using `xcodebuild` apparently doesn't count**.

This means that currently, Patrol tests can be run with `xcodebuild` works when in Profile or Release mode.

See also:
- https://github.com/flutter/engine/pull/20980#issue-692224079
- [flutter/engine - comment in the runtime/ptrace.cc file](https://github.com/flutter/engine/blob/6a92731d2edcb72dca0dc161de1da1e20a8210d8/runtime/ptrace_check.cc#L5-L21) – looks like we might be able to use `ios-deploy` to fix this

**Comment by bartekpacia** [comment]
On the good side, tests can be run on Firebase Test Lab even in Profile or Release mode (in fact, that might be preferable) on iOS.

Note for the future: the app has to be signed with any certificate (Developer certificate is fine). But if it's not signed at all, like this:


```
xcodebuild build-for-testing \
	-workspace Runner.xcworkspace \
	-scheme Runner \
	-xcconfig Flutter/Release.xcconfig \
	-configuration Release \
	-sdk iphoneos \
	-derivedDataPath ../build/ios_integ \
	OTHER_SWIFT_FLAGS='$(inherited) -D PATROL_ENABLED'
	CODE_SIGN_IDENTITY="" \
	CODE_SIGNING_REQUIRED="NO" \
	CODE_SIGN_ENTITLEMENTS="" \
	CODE_SIGNING_ALLOWED="NO"
```

the Firebase Test Lab won't work (failed run + "Some test executions didn't run any test cases").

**Comment by bartekpacia** [comment]
Fixed by #863
